src/pset/mod.rs

```diff
@@ -248,13 +253,296 @@ impl PartiallySignedTransaction {
+                self.outputs[i].amount = txout.value;
+                self.outputs[i].asset = txout.asset;
+                self.outputs[i].ecdh_pubkey = txout.nonce.commitment().map(|pk| bitcoin::PublicKey{
+                    key: pk,
+                    compressed: true
+                });
+            }
+            // return blinding factors used
+            ret.push((abf, vbf));
+        }
+
+        let (value, abf, vbf) = *out_secrets.last().unwrap();
+
+        // Calculate what should have been the last vbf if the txout
+        // had to be balanced
+        let vbf2 = ValueBlindingFactor::last(
+            secp,
+            value,
+            abf,
+            &inp_secrets,
+            &out_secrets[..out_secrets.len() - 1],
+        );
+
+        // Since the txout is not balanced, calculate the last scalar
+        let diff_scalar = {
+            if vbf2.into_inner() == ZERO_TWEAK {
+                vbf
+            } else if vbf.into_inner() == ZERO_TWEAK {
+                vbf2
+            } else {
+                // safe to turn into secret key
+                let sk2 = SecretKey::from_slice(vbf2.into_inner().as_ref()).expect("Valid key");
+                let mut sk = SecretKey::from_slice(vbf.into_inner().as_ref()).expect("Valid key");
+
+                sk.negate_assign();
+                sk.add_assign(sk2.as_ref())
+                    .expect("Random tweak is 0. Improbable event");
+                ValueBlindingFactor::from_slice(sk.as_ref()).expect("Valid Tweak")
+            }
+        };
+        // Push the scalar
+        // BUG in pset
+        // Bug in Pset, scalars can be the same value, but there is no place
+        // in pset to place them as it would break the uniqueness constriant.
+        self.global.scalars.push(diff_scalar.into_inner());
+        Ok(ret)
+    }
+
+    /// Blind the pset as the last blinder role. The non-last blinder of pset
+    /// should call the [`Self::blind_non_last`] function.
+    /// This function balances the blinding factors with partial information about
+    /// blinding inputs and scalars from [`Global`] scalars field.
+    /// `inp_secrets` and `out_secrets` must be consistent by [`Output`] `blinder_index` field
+    /// For each output, the corresponding inp_secrets\[out.blinder_index\] must be present
+    /// # Parameters
+    ///
+    /// * `inp_secrets`: [`TxOutSecrets`] corresponding to owned inputs. Use [`None`] for non-owned outputs
+    ///
+    pub fn blind_last<C: secp256k1_zkp::Signing, R: RngCore + CryptoRng>(
+        &mut self,
+        rng: &mut R,
+        secp: &secp256k1_zkp::Secp256k1<C>,
+        inp_txout_sec: &[Option<&TxOutSecrets>],
+    ) -> Result<(), PsetBlindError> {
+        let (_inp_secrets, mut outs_to_blind) = self.blind_checks(inp_txout_sec)?;
+
+        // If there are more than 1 outs to blind
+        // blind all outs but the last one
+        let last_out_index = outs_to_blind.pop().unwrap();
```

Comment:
...and if `blind_checks` didn't error out on "nothing to blind", this `unwrap` needs to be an error return

Reply:
See comment above